controllers/provisioning_controller.go

```diff
@@ -639,5 +664,6 @@ func (r *ProvisioningReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		Owns(&osconfigv1.ClusterOperator{}).
 		Owns(&osconfigv1.Proxy{}).
 		Owns(&machinev1beta1.Machine{}).
+		Watches(&source.Kind{Type: &corev1.Secret{}}, handler.EnqueueRequestsFromMapFunc(watchOCPConfigPullSecret)).
```

Comment:
AFAICT it's either that, or it's a completely new (independent) controller to watch the secret inside openshift-config

Reply:
This appears to watch every Secret in every namespace, which unfortunately (a) is pretty unsafe (we'll end up caching every user's Secret data in CBO), and (b) will 100% for sure chew up all of the RAM and cause the cluster to fall over, not necessarily in that order.

If we can't control this here, we need to control the scope in the Cache builder.

A related (but different) example: https://github.com/openshift/baremetal-operator/blob/master/main.go#L207-L209

Reply:
Hey, thanks for the comment.

The security issue isn't a problem of my commit, but of the following line, which literally gives the cluster-baremetal-operator the right to do anything with any configmaps, secrets or services in the entire cluster:
https://github.com/openshift/cluster-baremetal-operator/blob/5b91a35afaf62c59c289c85bd561a7fa308352ce/controllers/provisioning_controller.go#L106

For the performance concern, check main.go:
```
-	        Namespace:          controllers.ComponentNamespace,
+		NewCache: cache.MultiNamespacedCacheBuilder(
			[]string{controllers.ComponentNamespace, provisioning.OpenshiftConfigNamespace}),
```
The operator used to only watch resources in its own namespace, and with the change, should only watch resources in the openshift-config and its own namespace.

But I'll do some further testing with the rebased commit once I manage to spawn a cluster with clusterbot

Reply:
> the following line, which literally gives the cluster-baremetal-operator the right to do anything with any configmaps, secrets or services in the entire cluster

Unfortunately it has to be a superset of the RBAC roles it is deploying, so this is unavoidable.

> The operator used to only watch resources in its own namespace, and with the change, should only watch resources in the openshift-config and its own namespace.

Thanks, I indeed missed this. That should be sufficient AIUI.